File: admin/src/lib/slugs.js
// Slugify en JS (misma lógica que el backend: sin acentos, guiones)
export function slugify(texto) {
  if (!texto) return ''
  return texto
    .normalize('NFD')                          // separa diacríticos (á -> a+´)
    .replace(/[\u0300-\u036f]/g, '')           // elimina los diacríticos
    .replace(/[^a-zA-Z0-9\s-]/g, '')           // solo alfanuméricos, elimina puntos y rarezas
    .toLowerCase()
    .trim()
    .replace(/[\s-]+/g, '-')                   // espacios -> guiones colapsados
    .replace(/^-+|-+$/g, '')                   // sin guiones al inicio/final
    .slice(0, 330)                             // límite del backend (Article.slug 330)
}

File: backend/app/utils/slugs.py
# === Utilidades de slugs (transliteración de acentos al estilo URL) ===

import re
import unicodedata


def slugify(texto: str) -> str:
    """Convierte un texto a slug URL-friendly: minúsculas, sin acentos, guiones.

    Ejemplo: "Economía y Negocios" -> "economia-y-negocios"
    """
    if not texto:
        return ""
    # Normaliza y elimina diacríticos (á -> a)
    texto = unicodedata.normalize("NFKD", texto)
    texto = "".join(c for c in texto if not unicodedata.combining(c))
    # Minúsculas, solo alfanuméricos y guiones
    texto = re.sub(r"[^a-zA-Z0-9\s-]", "", texto).lower().strip()
    # Colapsa espacios y guiones repetidos y limpia bordes
    texto = re.sub(r"[\s-]+", "-", texto).strip("-")
    return texto[:330]
